src/main/java/jenkins/plugins/horreum/BaseExecutionContext.java

```diff
@@ -60,6 +61,10 @@ public BaseExecutionContext(String url, String credentials, PrintStream logger)
       );
       if (foundCredentials instanceof UsernamePasswordCredentials) {
          usernameCredentials = (UsernamePasswordCredentials) foundCredentials;
+         keyCredentials = null;
+      } else if (foundCredentials instanceof StringCredentials) {
+         usernameCredentials = null;
+         keyCredentials = (StringCredentials) foundCredentials;
       } else {
          throw new IllegalStateException("Could not retrieve Horreum Credentials. Please check the Horreum plugin configuration in Global Settings");
       }
```

Comment:
Here we are retrieving the credentials based on the concrete type.. but this could conflict with the authentication mechanism that has been configured. I would rather try to get the credentials based on the auth typ, e.g., if `API_KEY` I will try to retrieve a StringCredential, if not of that type I will raise an error (the same for `OIDC` or `BASIC` but with `usernameAndPassword` credential)

This way, if this check passes we are sure that either `keyCredentials` or `usernameCredentials` are populated if the corresponding auth type is enabled